ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -546,6 +546,16 @@ private static FormatInfo addSerializerMissingFormatFields(
           }
         }
       }
+
+      // If SCHEMA_ID was not added in the previous step, then use the SUBJECT_NAME
+      // Using a SUBJECT_NAME will help the serializer to identify which schema to fetch from SR
+      // but without using the restrictions we have with SCHEMA_ID
+      if (supportedProperties.contains(ConnectProperties.SUBJECT_NAME)) {
+        if (!formatInfo.getProperties().containsKey(ConnectProperties.SCHEMA_ID)) {
+          propertiesBuilder.put(ConnectProperties.SUBJECT_NAME,
```

Comment:
Is `subject_name`'s purpose to distinguish if protobuf serde is called from `InsertValuesExecutor`? Not sure if there's other ways...

Reply:
Yes. That's the only reason I added this. The other way was to pass the `topicName` when creating the Serde, but that causes too many refactor classes and test classes. I ended up just passing this info on the FormatInfo.